File: 11盛最多水的容器/src/Solution.java
public class Solution {
    public int maxArea(int[] height) {
        int left =0;
        int right = height.length-1;
        int h = Math.min(height[left], height[right]);
        int x = right - left;
        int ans = h * x;
        while (left < right){
            if(height[left] <= height[right]){
                left++;
            }else {
                right--;
            }
            h = Math.min(height[left], height[right]);
            x = right - left;
            ans = Math.max(ans, h*x);
        }

        return ans;
    }
}

File: 11盛最多水的容器/src/Solution.cpp
//
// Created by Frank on 2025/10/16.
//
#include <vector>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0;
        int left = 0;
        int right = height.size() - 1;
        while (left <= right) {
            int h = min(height[left], height[right]);
            int x = right - left;
            ans = max (ans, h * x);
            if (height[left] < height[right]) {
                left ++;
            }else {
                right --;
            }
        }
        return ans;
    }
};
